### ISSUE TITLE
bug with `async.filter` and `async.reject`

### ISSUE THREAD

**Issue by cvlmtg** [suguru03/neo-async#14]
Hi, I've a function that stats some files. during tests I stub that function so it no longer is asyncronous. with neo-async 1.2.1 the tests work correctly, but with async 1.3.1 the tests give an error.

here's a small coffee program that is capable of replicating the problem

```coffee
async = require 'neo-async'

names = [ 'dir1', 'dir2', 'file1', 'file2' ]

stubbedFunction = (name, callback) ->
  result = name.charAt(0) is 'd'
  callback null, result

tasks =
  dirs:  async.filter.bind null, names, stubbedFunction
  files: async.reject.bind null, names, stubbedFunction

async.parallel tasks, (err, result) ->
  console.log result
```

```sh
  ~ coffee test.coffee
{ dirs: [ 'dir1', 'dir2' ], files: [ 'file1', 'file2' ] }
  ~ npm uninstall neo-async
unbuild neo-async@1.2.1
  ~ npm install neo-async
neo-async@1.3.1 node_modules/neo-async
  ~ coffee test.coffee
Error: Callback was already called.
  at throwError (/Users/matteo/test/node_modules/neo-async/lib/async.js:9:11)
  at /Users/matteo/test/node_modules/neo-async/lib/async.js:7076:13
  at commonFilter (/Users/matteo/test/node_modules/neo-async/lib/async.js:3400:30)
  at baseEachFunc (/Users/matteo/test/node_modules/neo-async/lib/async.js:2691:20)
  at Object.parallel (/Users/matteo/test/node_modules/neo-async/lib/async.js:7067:9)
```

**Comment by suguru03** [comment]
@cvlmtg 
Thank you very much for your issue.
I found bug of `async.filter` and `async.reject`.
I'm going to fix as soon as possible and publish.

**Comment by suguru03** [comment]
I have just released v1.3.2! :smile: 
Would you check your test case?

**Comment by cvlmtg** [comment]
yes, it works :)
thanks!
